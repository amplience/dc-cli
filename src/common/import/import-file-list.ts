import fs from 'fs';

export interface ImportFile<O = {}> {
  uri: string;
  options: O;
}

export const getImportFileList = <T>(filePath: string): ImportFile<T>[] => {
  try {
    return JSON.parse(fs.readFileSync(filePath, 'utf-8'));
  } catch (err) {
    throw new Error(`Unable to read manifest file list: ${filePath}`);
  }
};
