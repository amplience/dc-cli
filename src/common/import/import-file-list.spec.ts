import fs from 'fs';
import { getImportFileList } from './import-file-list';

jest.mock('fs');

describe('import-file-list', () => {
  describe('getImportFileList', () => {
    afterEach((): void => {
      jest.resetAllMocks();
    });
    it('should return a file list containing a single file', () => {
      const manifest = [{ uri: '//remote-uri' }];
      (fs.readFileSync as jest.Mock).mockReturnValue(JSON.stringify(manifest));
      const result = getImportFileList('./manifest.json');
      expect(fs.readFileSync).toHaveBeenCalledTimes(1);
      expect(fs.readFileSync).toHaveBeenCalledWith('./manifest.json', 'utf-8');
      expect(result).toEqual(manifest);
    });

    it('should throw an error when manifest file contains invalid json', async () => {
      const manifest = 'invalid json';
      (fs.readFileSync as jest.Mock).mockReturnValue(manifest);
      expect(() => getImportFileList('./manifest.json')).toThrowErrorMatchingSnapshot();
    });

    it('should throw an error when manifest does not exist', async () => {
      (fs.readFileSync as jest.Mock).mockImplementationOnce(() => {
        throw new Error('File does not exist');
      });
      expect(() => getImportFileList('./manifest.json')).toThrowErrorMatchingSnapshot();
    });
  });
});
