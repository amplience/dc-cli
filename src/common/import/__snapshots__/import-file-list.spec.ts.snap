// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import-file-list getImportFileList should throw an error when manifest does not exist 1`] = `"Unable to read manifest file list: ./manifest.json"`;

exports[`import-file-list getImportFileList should throw an error when manifest file contains invalid json 1`] = `"Unable to read manifest file list: ./manifest.json"`;
